sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+                            }
+                            else
+                            {
+                                Debug.Assert(functionReturnType!="()","Returning value in function declared with no return type");
+                                var returnedVal = ret.Expression.RemoveParentheses();
+                                if (semanticModel.GetTypeInfo(returnedVal).Type == null &&
+                                    returnedVal.Kind() == SyntaxKind.SimpleMemberAccessExpression)
+                                {
+                                    // Special case a returning a method group that will be cast into a func
+                                    writer.WriteLine($"%{OpId(ret)} = cbde.unknown : none {GetLocation(ret)} // return method group");
+                                    writer.WriteLine($"return %{OpId(ret)} : none {GetLocation(ret)}");
+                                    break;
+                                }
+
+                                var id = ComputeCompatibleId(ret.Expression, functionReturnType);
+                                writer.WriteLine($"return %{id} : {functionReturnType} {GetLocation(ret)}");
+                            }
+                            break;
+                        case BreakStatementSyntax breakStmt:
+                            writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(breakStmt)} // break");
+                            break;
+                        case ContinueStatementSyntax continueStmt:
+                            writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(continueStmt)} // continue");
+                            break;
+                        case ThrowStatementSyntax throwStmt:
+                            // TODO : Should we transfert to a catch block if we are inside a try/catch?
+                            writer.WriteLine($"cbde.throw %{OpId(throwStmt.Expression)} :  {MLIRType(throwStmt.Expression)} {GetLocation(throwStmt)}");
+                            break;
+                        default:
+                            Debug.Assert(false, "Unknown kind of JumpBlock");
+                            break;
+                    }
+                    break;
+                case BinaryBranchBlock bbb:
+                    var cond = GetCondition(bbb);
+                    if (null == cond)
+                    {
+                        Debug.Assert(bbb.BranchingNode.Kind() == SyntaxKind.ForStatement);
+                        writer.WriteLine($"br ^{BlockId(bbb.TrueSuccessorBlock)}");
+                    }
+                    else
+                    {
+                        var id = EnforceBoolOpId(cond as ExpressionSyntax);
+                        writer.WriteLine($"cond_br %{id}, ^{BlockId(bbb.TrueSuccessorBlock)}, ^{BlockId(bbb.FalseSuccessorBlock)} {GetLocation(cond)}");
+                    }
+                    /*
```

Comment:
please remove the commented code and eventually add it to a CBDE ticket

Reply:
Move it to a non-code comment that indicates how the code works.